data_pipeline_api/registry/access_upload.py

```diff
@@ -298,12 +243,82 @@ def unique_posts(posts: List[YamlDict]) -> List[YamlDict]:
     return [yaml.safe_load(t) for t in set_of_yamls.keys()]
 
 
+def _get_accessibility(config):
+    if all(int(event.get("accessibility", 0)) == 0 for event in config["io"]):
+        return 0
+    else:
+        return max(int(event.get("accessibility", 0)) for event in config["io"])
+
+
+def _add_code_repo(posts: List[YamlDict], model_name: str, model_git_sha: str, model_uri: str) -> YamlDict:
+    """
+    Creates a code_repo_release and adds it to the list of objects to post to the data registry
+
+    :param posts: List of posts to the data registry, will be modified
+    :param model_name: name of the model
+    :param model_git_sha: git sha of the model
+    :param model_uri: uri that the model is located at
+    :return: the object reference to the code repo release
+    """
+    code_repo_obj = _create_target_data_dict(
+        DataRegistryTarget.object, {DataRegistryField.description: f"{model_name}+{model_git_sha}"}
+    )
+    code_repo_release = _create_target_data_dict(
+        DataRegistryTarget.code_repo_release,
+        {
+            DataRegistryField.name: model_name,
+            DataRegistryField.version: model_git_sha,
+            DataRegistryField.website: model_uri,
+            DataRegistryField.object: code_repo_obj,
+        },
+    )
+    posts.append(code_repo_release)
+    posts.append(code_repo_obj)
+    return code_repo_obj
+
+
+def _upload_file_to_storage(
+    posts: List[YamlDict],
+    filename: Union[str, Path],
+    remote_uri: str,
+    storage_options: Dict[str, str],
+    storage_root: Union[str, YamlDict],
+) -> YamlDict:
+    """
+    for a given filename, uploads it to the remote uri and returns a reference to the object that will be posted
+
+    :param posts: List of posts to the data registry, will be modified
+    :param filename: path to the file to upload
+    :param remote_uri: URI to the root of the storage for uploading
+    :param storage_options: (key, value) pairs that are passed to the remote storage, e.g. credentials
+    :param storage_root: existing reference to the storage_root that this was uploaded to
+    :return: object reference to the uploaded file
+    """
+    filename = Path(filename)
+    path = upload_to_storage(remote_uri, storage_options, filename.parent, filename)
+    with open(filename, "rb") as file:
```

Comment:
You had mentioned using the FileAPI version of this, or abstracting that out. I don't really mind if you don't, though.